src/Vts.Scripting/ShortCourse/Demo01APhotonCountWithFluence.cs

```diff
@@ -0,0 +1,124 @@
+                    zRange: new(100, double.PositiveInfinity),       // air "z" range
+                    op: new(mua: 0.0, musp: 1E-10, g: 1.0, n: 1.0))  // air optical properties
+            }
+        };
+
+        // define a single fluence(rho,z) detector by the endpoints of rho and z bins
+        var rhoRange = new DoubleRange(start: 0, stop: 10, number: 101);
+        var zRange = new DoubleRange(start: 0, stop: 10, number: 101);
+        var detectorInputs = new IDetectorInput[]
+        {
+            new FluenceOfRhoAndZDetectorInput { Rho = rhoRange, Z = zRange, TallySecondMoment = true, Name = "FluenceOfRhoAndZ" }, // name can be whatever you want
+        };
+
+        // specify simulation options
+        var options = new SimulationOptions
+        {
+            Seed = 0, // -1 will generate a random seed
+            AbsorptionWeightingType = AbsorptionWeightingType.Analog,
+            PhaseFunctionType = PhaseFunctionType.HenyeyGreenstein
+        };
+
+        // define how many different photon counts to simulate
+        var numPhotonsArray = new[] { 10, 100, 1000, 10000 };
+
+        // create an array of simulations, one for each different photon count value
+        var allSimulations = numPhotonsArray.Select(n => 
+            new MonteCarloSimulation(
+                new SimulationInput(
+                    numberOfPhotons: n,
+                    outputName: "results",
+                    simulationOptions: options,
+                    sourceInput: sourceInput,
+                    tissueInput: tissueInput,
+                    detectorInputs: detectorInputs)
+            )).ToArray();
+
+        // run all the simulations in parallel
+        var allSimulationOutputs = MonteCarloSimulation.RunAll(allSimulations);
+
+        // collect all the results in helpful arrays
+        var allFluenceDetectors = allSimulationOutputs.Select(detectorResult => 
+            (FluenceOfRhoAndZDetector)detectorResult.ResultsDictionary["FluenceOfRhoAndZ"]).ToArray();
+        var allFluenceMeans = allFluenceDetectors.Select(detector => detector.Mean.ToEnumerable<double>().ToArray()).ToArray();
+        var allFluenceSecondMoments = allFluenceDetectors.Select(detector => detector.SecondMoment.ToEnumerable<double>().ToArray()).ToArray();
+
+        // compute the relative error (standard deviation / mean) for each simulation
+        var allRelativeErrors = numPhotonsArray.Select((numPhotons, npIdx) =>
+            allFluenceMeans[npIdx].Zip(allFluenceSecondMoments[npIdx], (mean, secondMoment) => 
+                Math.Sqrt((secondMoment - mean * mean) / numPhotons) / mean).ToArray()).ToArray();
+
+        // plot the results using Plotly.NET
+        var rhos = rhoRange.GetMidpoints();
+        var zs = zRange.GetMidpoints();
+        var allRhos = rhos.Select(rho => -rho).Reverse().Concat(rhos).ToArray(); // duplicate for -rho to make symmetric
+        var charts = numPhotonsArray.Select((numPhotons, npIdx) =>
```

Comment:
numPhotons is not used replace with _?

Reply:
This (and all similar examples) is an optimization that doesn't help someone using this code as a jumping off point. In the general case, they'll want access to the main variable they're looping over.

Reply:
That's what I figured, just being thorough in my review :)